src/tqec/compile/tree/node.py

```diff
@@ -131,27 +131,13 @@ def get_annotations(self, k: int) -> LayerNodeAnnotations:
     def set_circuit_annotation(self, k: int, circuit: ScheduledCircuit) -> None:
         self.get_annotations(k).circuit = circuit
 
-    def generate_circuit(
+    def _generate_circuits_with_potential_polygons(
```

Comment:
I think this method can be public (and should at least document its returned value)